### PR TITLE
Chore: Artifact collections

### PR DESCRIPTION
**Pull request by znicholasbrown**
This PR adds more explicit support for artifact collections (containing objects for the `latest` artifact of a given key) and drops support for `is_latest` in the artifacts filter route. Updates all relevant components to use the new route

### CODE REVIEW

**Comment by znicholasbrown** [comment]
Ah I just realized this'll break the mock api - let me update that quick

File: src/components/ArtifactTimeline.vue [hunk 1]
```diff
@@ -119,7 +112,8 @@
       expanded.value = [val]
     }
     getArtifacts()
-  })
+  }, { immediate: true })
```
Comment (pleek91) [inline src/components/ArtifactTimeline.vue:115]: You're calling getArtifacts in the onBeforeMount and then again here in this watcher. Adding immediate means it's going to get artifacts twice. Is that intentional?

**Review by pleek91 — APPROVED** [review]
Everything looks like I'd expect. One thing about the watcher that doesn't seem quite right but approving in case that's intentional or you want to fix in a future PR

File: src/components/ArtifactTimeline.vue (hunk 1 above)
Comment (znicholasbrown) [reply src/components/ArtifactTimeline.vue:115]: Ah actually that's an oversight; we should only refetch artifacts if this _isn't_ the first iteration. Will update, nice catch

**Pull request merged**